sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroup.java

```diff
@@ -0,0 +1,233 @@
+        return this.targetThroughput;
+    }
+
+    /**
+     * Set the throughput control group target throughput. The value should be larger than 0.
+     *
+     * @param targetThroughput the target throughput for the throughput control group.
+     * @return the {@link ThroughputControlGroup}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup targetThroughput(int targetThroughput) {
+        checkArgument(targetThroughput > 0, "Target throughput should be larger than 0");
+        this.targetThroughput = targetThroughput;
+        return this;
+    }
+
+    /**
+     * Get the throughput control group target throughput threshold.
+     *
+     * @return the target throughput threshold.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Double getTargetThroughputThreshold() {
+        return this.targetThroughputThreshold;
+    }
+
+    /**
+     * Set throughput control group target throughput threshold. The value should be larger than 0.
+     *
+     * @param targetThroughputThreshold the target throughput threshold for the throughput control group.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup targetThroughputThreshold(double targetThroughputThreshold) {
+        checkArgument(
+            targetThroughputThreshold > 0 && targetThroughputThreshold <= 1,
+            "Target throughput threshold should between (0, 1]");
+
+        this.targetThroughputThreshold = targetThroughputThreshold;
+        return this;
+    }
+
+    /**
+     * Get whether this throughput control group will be used by default.
+     *
+     * By default, it is false.
+     *
+     * @return {@code true} this throughput control group will be used by default unless being override. {@code false} otherwise.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public boolean isUseByDefault() {
+        return this.useByDefault;
+    }
+
+    /**
+     * Set the throughput control group to be used by default.
+     *
+     * @return the {@link ThroughputControlGroup}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ThroughputControlGroup useByDefault() {
+        this.useByDefault = Boolean.TRUE;
+        return this;
+    }
+
+    /**
+     * Get the id of the throughput control group.
+     *
+     * @return the throughput control group id.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public String getId() {
+        if (StringUtils.isEmpty(this.id)) {
+            this.validate();
+            this.id = String.format(
+                "%s.%s.%s",
```

Comment:
this uses regex underneath which is expensive. If it is used on the hot path it is better to replace with string concat or string builder.

Reply:
updated all the above comments.